### ISSUE TITLE
Only the last compiled .proto file output is included in "qtprotobuf_global.qpb.h"

### ISSUE THREAD

**Issue by R0ll1ngSt0ne** [semlanik/qtprotobuf#28]
It seems that only the class output of only the last compiled .proto file is included in "qtprotobuf_global.qpb.h". For example, if the project contains Test1.proto and Test2.proto, during build I see the following:

Compiling Protobuf schema Test1.proto (Qt)
Compiling Protobuf schema Test2.proto (Qt)

So, protoc is executed twice - once for each file. The output ""qtprotobuf_global.qpb.h" file contains the following (note that Test1 is missing):

> /* This file is autogenerated. DO NOT CHANGE. All changes will be lost */
> 
> #pragma once
> 
> #include <QObject>
> #include <QMetaType>
> #include <QList>
> #include <QProtobufObject>
> #include <QSharedPointer>
> 
> #include <memory>
> #include <unordered_map>
> 
> #include "Test2.qpb.h"
> 
> namespace nvram {
> inline void qRegisterProtobufTypes() {
>     qRegisterProtobufType\<Test2\>();
> }
> }

**Comment by semlanik** [comment]
@R0ll1ngSt0ne is it possible from your side to use single-call generation? I know that dependency resolving is not complete yet. But probably you could use:

```cmake
generate_qtprotobuf(TARGET ${TARGET}
    OUT_DIR ${CMAKE_CURRENT_BINARY_DIR}/generated
    PROTO_FILES full/path/to/Test1.proto 
                             full/path/to/Test2.proto 
    QML TRUE)
```

I'm not sure if it's feasible to fix this issue on generator level at all. As you can see for well-known-types, that has similar issue with multi-generation we made separate qtprotobufwellknowntypes_global.qbp.h. The issue why we could not use append at this point is because at generation time there might be result of previous generations.

So suggestion for now is to use single generation call. I will keep issue open for now.

**Comment by R0ll1ngSt0ne** [comment]
We do not use CMAKE to process .proto files - direct "protoc" call is issued from Visual Studio for each one at compile time. So, the option above is not an option for us.

**Comment by semlanik** [comment]
@R0ll1ngSt0ne, than as solution you probably may use 2 different folders as generator output. It will solve your issue

**Comment by R0ll1ngSt0ne** [comment]
@semlanik, the best solution for this issue can be found in the suggestion for Qt Protobuf type registration I just submitted as a separate ticket #55. Please take a look.

**Comment by semlanik** [comment]
Will be fixed by #55
